pkg/rancher-desktop/config/__tests__/settings.spec.ts

```diff
@@ -249,104 +146,99 @@ describe('settings', () => {
-        <dict>
-          <key>title</key>
-          <string>sub-title</string>
-          <key>locked</key>
-          <true/>
-          <key>subvar</key>
-          <string>sub-var</string>
-        </dict>
-      </dict>
-    </dict>
-    <key>debug</key>
-    <true/>
-    <key>diagnostics</key>
-    <dict>
-      <key>showMuted</key>
-      <false/>
-      <key>locked</key>
-      <true/>
-      <key>mutedChecks</key>
-      <dict/>
-    </dict>
-  </dict>
-</plist>
-  `;
-  // TODO: Stop doing this once profiles are implemented on windows
+  };
+  const unlockedJSONProfile = JSON.stringify(unlockedProfile);
+  const lockedJSONProfile = JSON.stringify(lockedProfile);
+  const unlockedPlistProfile = plist.build(unlockedProfile);
+  const lockedPlistProfile = plist.build(lockedProfile);
+
+  // Check structural breakage in this file.
+  const brokenJSONProfile = jsonProfile.slice(0, jsonProfile.length / 2);
+  const brokenPlistProfile = plistProfile.slice(0, plistProfile.length / 2);
+
+  // TODO: Figure out how to implement this on Windows as well
   const describeNotWindows = process.platform === 'win32' ? describe.skip : describe;
 
-  describe('profiles', () => {
-    describeNotWindows('locked fields', () => {
-      const lockedAccessors = ['containerEngine.allowedImages.enabled', 'containerEngine.allowedImages.patterns'];
-      let mock: jest.SpiedFunction<typeof fs['readFileSync']>;
-      const actualSyncReader = fs.readFileSync;
+  describeNotWindows('profiles', () => {
+    const lockedAccessors = ['containerEngine.allowedImages.enabled', 'containerEngine.allowedImages.patterns'];
+    let mock: jest.SpiedFunction<typeof fs['readFileSync']>;
+    const actualSyncReader = fs.readFileSync;
+
+    beforeEach(() => {
+      jest.spyOn(fs, 'writeFileSync').mockImplementation(() => { });
+    });
+    afterEach(() => {
+      mock.mockRestore();
+    });
+    function createMocker(useSystemProfile: ProfileTypes, usePersonalProfile: ProfileTypes, typeToCorrupt = ''): (inputPath: any, unused: any) => any {
+      return (inputPath: any, unused: any): any => {
+        if (!inputPath.startsWith(paths.deploymentProfileUser) && !inputPath.startsWith(paths.deploymentProfileSystem)) {
+          return actualSyncReader(inputPath, unused);
+        }
+        const action = inputPath.startsWith(paths.deploymentProfileSystem) ? useSystemProfile : usePersonalProfile;
 
+        if (action === ProfileTypes.None || inputPath === path.join(paths.config, 'settings.json')) {
+          throw new FakeFSError(`File ${ inputPath } not found`, 'ENOENT');
+        }
+        const pathInfo = path.parse(inputPath);
+
+        if (!['.json', '.plist'].includes(pathInfo.ext)) {
+          if (pathInfo.base === 'settings.json') {
+            throw new FakeFSError(`File ${ inputPath } not found`, 'ENOENT');
+          }
+
+          return actualSyncReader(inputPath, unused);
+        }
+
+        if (inputPath.endsWith('defaults.json')) {
```

Comment:
Actually, we need to use the actual basenames, `rancher-desktop.{defaults,locked}.json` on linux and `io.rancherdesktop.profile.{defaults,locked}.plist` on macOS. I was trying to avoid that. Guess not.

Reply:
Ah, yes, please add those comments; that was unclear. (But that means we can't compare with `.json` because `plist`.)